### ISSUE TITLE
[Feature Request] RetroAchievements support?

### ISSUE THREAD

**Issue by Blakeinstein** [rileytestut/Delta#196]
One reason I still have to use the inferior looking retroarch (awesome on consoles) is because of retro achievements.

With internal delta cores using melonds and the like, would it be possible in the ever future to integrate support for RA in delta?

For those who are not aware, retro achievements is an achievement tracker for retro games! Learn more here https://retroachievements.org/

**Comment by lonkelle** [comment]
This is one of my top priorities for new features rather than bug fixes, I absolutely love RetroAchievements. ❤️

**Comment by nighto** [comment]
One year later, could we hope for an eventual RetroAchievements support? I'd love to play NDS with RA but unfortunately the only mobile option is in Android with melonDS. Thanks in advance :)

**Comment by saulojoab** [comment]
Any news on this? Would LOVE to play GBA and NDS games with RetroAchievements compatibility.

**Comment by tlrwagner** [comment]
Especially with the AppStore release now, it would be cool to see retro achievement support

**Comment by adamstirtan** [comment]
I’ve played most of these games so many times now. Retroachievements is really the only reason I’d want to play Super Mario World for the 700th time. This project is awesome and if it had RA support, I’d never get any work done.

**Comment by mtoensing** [comment]
I second this! RetroAchievements are the only thing that is missing to ditch my Anbernic Handhelds for my iPhone. And RetroArch is not available on AltStore for me.

**Comment by nighto** [comment]
@mtoensing you can get a RetroArch .ipa on their website and install it with AltStore manually. That's how I roll on my iPhone :)

**Comment by mtoensing** [comment]
Without the Mac app? Just by using my phone? Is there a link for a tutorial?

**Comment by nighto** [comment]
@mtoensing you need the Mac or Windows app to do the sign, just like you need it to keep resigning every week, but you can do the whole process on your phone. There is no need for a tutorial: when opening AltStore you should see a Plus button on the top left, you click there, pick up the .ipa file (for instance on Files app, after having downloaded it on your browser or copied somehow) and that's it.

**Comment by mtoensing** [comment]
@nighto I live in the EU and opted for the EU AltStore App. So no Mac app. I don’t see a plus button other than to add new sources in the AltStore app. And I am a Patreon.

**Comment by nighto** [comment]
@mtoensing Oh I see. Then I don't know, didn't try that yet, sorry.

**Comment by carriergrady** [comment]
Although I love Delta in its current state, I think the lack of RetroAchievements support is the only thing holding it back in my book. If such support is implemented, I think this might flat-out become my most used iPhone app. Especially if RetroAchivement's hardcore mode is implemented, the app would feel complete to me. Here's to hoping for RA support!

**Comment by adamstirtan** [comment]
> Although I love Delta in its current state, I think the lack of RetroAchievements support is the only thing holding it back in my book. If such support is implemented, I think this might flat-out become my most used iPhone app. Especially if RetroAchivement's hardcore mode is implemented, the app would feel complete to me. Here's to hoping for RA support!

I'm with you on this. This app is amazing but I'm simply not going to play unless RA is here.

**Comment by nighto** [comment]
@mtoensing just to let you know that RetroArch launched on App Store ;-)

**Comment by mtoensing** [comment]
@nighto thanks! I already installed at and now enjoy emulation on iOS with RA. Uninstalled delta since it is now obsolete

**Comment by adamstirtan** [comment]
![achievement](https://github.com/rileytestut/Delta/assets/9989813/0ecc776b-2e25-4ceb-a7d5-5c62087c909c)

**Comment by sevansup** [comment]
RetroAchievements is the only thing making me want to use RetroArch. The Delta user experience is so much better, but I don't feel like using it without achievement support...hope it gets added. If I'm gonna spend all this time playing old games, I'd like to at least be able to show my achievements. :)

**Comment by wescopeland** [comment]
RetroAchievements admin here 👋

We would love to see Delta support RetroAchievements one day. Please reach out to us if you need support. Additionally, we have an emulator integration guide [here](https://github.com/RetroAchievements/rcheevos/wiki/rc_client-integration).

**Comment by lemon5ky** [comment]
Any updates?

**Comment by spaceage64** [comment]
Yes there’s an update apparently:
https://www.threads.net/@rileytestut/post/DHTw5tVR343

Seems like it’s really happening 🙌

**Comment by Blakeinstein** [comment]
I just heard the news and immediately subbed for Patreon. Just realised there is only NES, SNES and n64 support.

Couldn't find commits implementing this support, would love to know how it was done and maybe contribute as-well!

**Comment by spaceage64** [comment]
Yes, NES, SNES, and N64 for now only, though the release notes do state that the [other systems are "coming soon."](https://faq.deltaemulator.com/release-notes#:~:text=Remaining%20systems%20coming%20soon!)

**Comment by solitonmedic** [comment]
Wait, it took this long to implement RA? I thought it already had it.

**Comment by Chromagram** [comment]
I'm honestly a bit worried for the progress of this. It's less about acknowledging that programming this probably takes a good amount of time and there's likely also a ladder of priority for which features come first, but the emulator gets one update log every like 3+ months right now, and the last one just made accessing the Patreon info easier.

For me, RA is the only way I play retro games right now. I have a PC/Steam Deck that comfortably handles most of the consoles, but Delta on an iPad with an Apple Pencil would truly be the only way to marry RetroAchievements support with a touch screen that have a proper pointed tip stylus and super low latency for the very precision-based gesture games like Trauma Center, Rhythm Heaven, The World Ends With You, etc.

RA's been sorely lacking the ability to play DS games optimally and Delta releasing support for DS would 1000% pioneer this system's support on the entire site. I'm really hoping for some news on DS RA support.

<img width="1022" height="131" alt="Image" src="https://github.com/user-attachments/assets/931d8e3d-8d32-40a1-9b22-5c5e0783e115" />
For context this is what was posted by the RA admin directly as part of a site update. They say they've reached out to the devs on just a base timeline but they're currently set to be delisted from hardcore achievements support by tomorrow with no info.

Not posting this to be upset or critical in any way, just my general concern. If any Delta dev/admin picks up on this, please keep the support alive. Playing RA without hardcore achievements support is like ordering a root beer float without the ice cream. This could be the cleanest, easiest and absolute best way to play DS RA too if supported.

**Comment by rholliday** [comment]
> 
> <img width="1022" height="131" alt="Image" src="https://github.com/user-attachments/assets/931d8e3d-8d32-40a1-9b22-5c5e0783e115" />
> For context this is what was posted by the RA admin directly as part of a site update. They say they've reached out to the devs on just a base timeline but they're currently set to be delisted from hardcore achievements support by tomorrow with no info.
> 
> Not posting this to be upset or critical in any way, just my general concern. If any Delta dev/admin picks up on this, please keep the support alive. Playing RA without hardcore achievements support is like ordering a root beer float without the ice cream. This could be the cleanest, easiest and absolute best way to play DS RA too if supported.


Locking RetroAchievements behind a paywall is a violation of their hardcore policy. I hope it is corrected. It also just doesn't make sense, RA is the kind of thing to drive traffic to an emulator. If you're going to lock a feature behind a paywall I'd assume it would be convenience stuff like syncing with Dropbox/Google Drive.

**Edit:** I found a post (somewhat ironically locked behind subscription to the Patreon) [that indicates RA will not be behind a paywall in the next version.](https://www.patreon.com/posts/139719692?utm_campaign=postshare_fan)

**Comment by nighto** [comment]
Copying the text (from September 26, 2025) for those without Patreon access:

> Hi Patrons,
>
>We wanted to give you a quick announcement about RetroAchievement support in Delta:
>
>When iOS 26 and Liquid Glass first got announced, we were excited by the opportunity to update Delta’s UI and make it more responsive and playful! But Liquid Glass has evolved a lot over the summer, and it took much more development time than we originally anticipated. Because of this, we decided to delay Delta 2.0 closer to the end of the year — which included the final version of RetroAchievements along with it.
>
>However, delaying Delta 2.0 caused an issue with our RetroAchievements support. Since our beta is only available for Patrons and RetroAchievements is a free, community-run service, they contacted us to give us a heads up that they'd be disabling Hardcore Mode for Delta. They explained this was to avoid the perception that RetroAchievements is a paid feature, and we completely understood. We didn't want to give that perception either, we just also didn't want to release it to everyone until it was finished. So in order to re-enable Hardcode Mode while still finalizing RetroAchievements support for 2.0, we’ve been working closely with the RetroAchievements team and found a way forward! We've made a TestFlight version of Delta that includes our WIP RetroAcheivements implementation, and you can join that here: https://testflight.apple.com/join/aFVsc4vY
>
>However, if you just want to keep using whatever version of Delta you have, we'll also be releasing updates to both the AltStore and App Store versions soon to re-enable Hardcore Mode. Stay tuned!
>
>We want to give a big thank you to the RetroAchievements team for all of their help on this, and we sincerely apologize for any inconvenience this might have caused in the meantime.
>
>Riley + Shane

**Comment by pacorob** [comment]
When can we expect the v2.0 release with RA?
